grpcx/deamon.go

```diff
@@ -0,0 +1,57 @@
+package grpcx
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"google.golang.org/grpc"
+
+	"github.com/socialpoint-labs/bsk/contextx"
+)
+
+// Daemon represents a runnable gRPC daemon
+type Daemon struct {
+	port int
+	svr  *grpc.Server
+	apps []Application
+}
+
+// Application is a contextx.Runner that can register gRPC services
+type Application interface {
+	contextx.Runner
+	RegisterGRPC(grpc.ServiceRegistrar)
+}
+
+func NewDaemon(port int, svr *grpc.Server, apps ...Application) Daemon {
+	return Daemon{
+		port: port,
+		svr:  svr,
+		apps: apps,
+	}
+}
+
+// Run registers and runs every application of the Daemon with a gRPC server,
+// then starts it and gracefully shuts it down once the context is done.
+func (d Daemon) Run(ctx context.Context) {
+	for _, app := range d.apps {
+		app.RegisterGRPC(d.svr)
+
+		go app.Run(ctx)
+	}
+
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", d.port))
+	if err != nil {
+		panic(err.Error())
```

Comment:
why not return an error? because I know that you want to panic here if you can up the listener, but, this is a library should not decide if do panic or not, should be the application that implement this who should be in charge of handle it if produce a panic or other way

Reply:
Here we're adhering to the `contextx.Runner` interface, which is the type of our runners (see https://github.com/socialpoint-labs/bsk/blob/master/contextx/runner.go#L8).

I changed the code replacing panics with an error func.